### PR TITLE
Add API to SFrameTransform to set the crypto key

### PR DESCRIPTION
**Pull request by youennf**
***
<a href="https://pr-preview.s3.amazonaws.com/youennf/webrtc-insertable-streams/pull/53.html" title="Last updated on Feb 10, 2021, 11:37 AM UTC (1116548)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-insertable-streams/53/f9e6dc1...youennf:1116548.html" title="Last updated on Feb 10, 2021, 11:37 AM UTC (1116548)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -219,7 +219,7 @@ Similarly, if a web application sets the transform synchronously at creation of
 
 # SFrameTransform # {#sframe}
 
-<pre class="idl">
+<xmp class="idl">
```
Comment (dontcallmedom) [inline index.bs:222]: is that intended?

Comment (youennf) [reply index.bs:222]: Without this change, Promise<undefined> triggers a build issue.

Comment (dontcallmedom) [reply index.bs:222]: oh, I wasn't aware of that [bikeshed trick](https://tabatkins.github.io/bikeshed/#xmp)

File: index.bs [hunk 2]
```diff
@@ -262,6 +263,15 @@ The SFrame transform algorithm, given |sframe| as a SFrameTransform object and |
 12. If |frame| is an {{RTCEncodedVideoFrame}}, set |frame|.{{RTCEncodedVideoFrame/data}} to |buffer|.
 13. [=ReadableStream/Enqueue=] |frame| in |sframe|.`[[transform]]`.
 
+## Methods
+The <dfn method for="SFrameTransform">setEncryptionKey(|key|, |keyID|)</dfn> method steps are:
+1. Let |promise| be [=a new promise=].
+2.  [=In parallel=], run the following steps:
+    1. Set |key| with its optional |keyID| as key material to use for the SFrame transform algorithm, as defined by the <a href="https://datatracker.ietf.org/doc/draft-omara-sframe/">SFrame specification</a>.
+    2. [=Resolve=] |promise| with undefined.
```
Comment (alvestrand) [inline index.bs:271]: "If setting the key fails, reject the promise with ......"

File: index.bs [hunk 3]
```diff
@@ -262,6 +263,15 @@ The SFrame transform algorithm, given |sframe| as a SFrameTransform object and |
 12. If |frame| is an {{RTCEncodedVideoFrame}}, set |frame|.{{RTCEncodedVideoFrame/data}} to |buffer|.
 13. [=ReadableStream/Enqueue=] |frame| in |sframe|.`[[transform]]`.
 
+## Methods
+The <dfn method for="SFrameTransform">setEncryptionKey(|key|, |keyID|)</dfn> method steps are:
+1. Let |promise| be [=a new promise=].
+2.  [=In parallel=], run the following steps:
+    1. Set |key| with its optional |keyID| as key material to use for the SFrame transform algorithm, as defined by the <a href="https://datatracker.ietf.org/doc/draft-omara-sframe/">SFrame specification</a>.
```
Comment (youennf) [inline index.bs:270]: Add a step to reject promise if the algorithm has an error

**Review by henbos — APPROVED** [review]

File: index.bs (hunk 3 above)
Comment (jan-ivar) [reply index.bs:270]: @youennf Looks like this got merged already. Should we open a followup issue for the error handling?

Comment (youennf) [reply index.bs:270]: Right, I just filed https://github.com/w3c/webrtc-insertable-streams/issues/56 for that purpose.

Comment (youennf) [reply index.bs:270]: Will file a PR soon

**Pull request merged**